### PR TITLE
[script] [forging] Adding support for adjustable tongs

### PR DESCRIPTION
**Pull request by Raykyn55**
Checks for adjustable tongs, if they adjust, uses them as shovel, then reverts before next crafting action. Also corrects tongs in wrong adjustment during pound.

### CODE REVIEW

**Comment by vtcifer** [comment]
II would recommend a new yaml setting to enable this, rather than assuming everyone uses adjustable tongs, otherwise there's a bit of a regression of functionality here.

**Comment by Raykyn55** [comment]
Still works for regular tongs

**Comment by SekmehtDR** [comment]
Nice job putting something together! I see where VTC is coming from. Could have it do a get_settings for swappable_tongs: true, then if true, then do the adjustment, if not then not even check it. 

I'd like to say most of everyone who has high end tongs, or covellite articulated tongs even, all of those are basically swappable, so the assumption part is semi-valid.

**Comment by Raykyn55** [comment]
OK, so I can make this a yaml setting, but didn't for a couple reasons (though absolutely willing to go that route)
1. The changes I made here trigger a quick attempt at adjustment maybe once per item, and if it fails, it just carries on getting a shovel. So no one loses function.
2. The yaml is already pretty big, and I didn't want to add more niche settings where something like this could suffice (IMO)
3. When testing, i discovered that use of the shovel is pretty rare(to my dismay, prob went through 20 hand claws to get 5 tests)

That said, I am working on a yaml setting version, how would I submit that? Put it in here?

**Comment by SekmehtDR** [comment]
I like the quick adjust and match with it, if it adjusts then it can be used as shovel then returned... Makes since to me right?

**Comment by MahtraDR** [comment]
> That said, I am working on a yaml setting version, how would I submit that? Put it in here?

You'd add it to this PR.

**Comment by SekmehtDR** [comment]
> 

Yeah and like Mahtra said, when you make any changes to say base.yaml, base will be listed as another file changed as long as it's all within the same branch :)

**Comment by SekmehtDR** [comment]
@MahtraDR in the title of this PR, maybe remove clockwork, because there's actually a lot of adjustable tongs in the game, the craft able one being articulated tongs. So I'd make it "... for adjustable tongs"

**Comment by MahtraDR** [comment]
> @MahtraDR in the title of this PR, maybe remove clockwork, because there's actually a lot of adjustable tongs in the game, the craft able one being articulated tongs. So I'd make it "... for adjustable tongs". But hell yeah im using my clockworks with it :D

I did not know there was more than one kind of adjustable tongs!

**Comment by SekmehtDR** [comment]
> I did not know there was more than one kind of adjustable tongs!

Cool right!?! swappable between shovel and tong and retains the speed!

File: forge.lic [hunk 1]
```diff
@@ -284,6 +284,7 @@ class Forge
       pound(item)
     when /You must be holding some metal tongs in your other hand to do that/
       get_item('tongs')
+      bput("adjust my tongs", 'alongside the tong', 'You cannot adjust')
```
Comment (MahtraDR) [inline forge.lic:287]:
```suggestion
      DRC.bput("adjust my tongs", 'alongside the tong', 'You cannot adjust')
```

File: forge.lic [hunk 2]
```diff
@@ -321,16 +322,28 @@ class Forge
...
-    get_item(@shovel)
-    case bput('push fuel with my shovel', 'Roundtime', 'That tool does not seem')
-    when 'That tool does not seem'
-      analyze_item
-    else
-      waitrt?
-      stow_item(@shovel)
-      get_item('tongs')
-      pound
+    case bput("adjust my tongs", 'alongside the tong', 'You cannot adjust', 'You have no idea how')
```
Comment (MahtraDR) [inline forge.lic:325]:
```suggestion
    case DRC.bput("adjust my tongs", 'alongside the tong', 'You cannot adjust', 'You have no idea how')
```

File: forge.lic [hunk 3]
```diff
@@ -321,16 +322,28 @@ class Forge
...
+    when 'You cannot adjust', 'You have no idea how'
+      stow_item('tongs')
+      get_item(@shovel)
+      case bput('push fuel with my shovel', 'Roundtime', 'That tool does not seem')
```
Comment (MahtraDR) [inline forge.lic:329]:
```suggestion
      case DRC.bput('push fuel with my shovel', 'Roundtime', 'That tool does not seem')
```

File: forge.lic [hunk 4]
```diff
@@ -321,16 +322,28 @@ class Forge
...
+      when 'That tool does not seem'
+        analyze_item
+      else
+        waitrt?
+        stow_item(@shovel)
+        get_item('tongs')
+        pound
+      end
+    when /alongside the tong/
+      case bput('push fuel with my tongs', 'Roundtime', 'That tool does not seem')
```
Comment (MahtraDR) [inline forge.lic:339]:
```suggestion
      case DRC.bput('push fuel with my tongs', 'Roundtime', 'That tool does not seem')
```

File: forge.lic [hunk 5]
```diff
@@ -321,16 +322,28 @@ class Forge
...
+      when 'That tool does not seem'
+        analyze_item
+      else
+        waitrt?
+        bput("adjust my tongs", 'alongside the tong')
```
Comment (MahtraDR) [inline forge.lic:344]:
```suggestion
        DRC.bput("adjust my tongs", 'alongside the tong')
```

**Review by MahtraDR — CHANGES_REQUESTED** [review]
Please use explicit prefixes for the common methods. I've suggested the change on all the bputs.

**Comment by Raykyn55** [comment]
Ok, added those DRCs, removed includes, and made it a yaml setting: adjustable_tongs

**Comment by Raykyn55** [comment]
false in base, blank in empty

**Comment by SekmehtDR** [comment]
`R> 
[forge]>pound fan on anvil with my forging hammer

After using the tongs to warm the fan over the forge fire, you place it down upon the anvil and make some adjustments to its shape with carefully timed hammer taps.  The forging proceeds exceptionally well and free of mistakes.

As you complete working the fire dies down and appears to need some more fuel.
Roundtime: 9 sec.

R> 
Gained: Forging(+1)

R> 
[forge]>adjust my tongs

You lock the tongs into a fully extended position.  Then you reach alongside the tong's body, and pull free the head of a shovel.

> 
[forge]>push fuel with my tongs

Gripping the tongs tightly, you slam it deep into the pile of fuel and give it a hard kick.   Straining, you remove the tongs with a mound of coal atop it and toss the coal onto the fire.
Roundtime: 10 sec.

R> 
[forge]>adjust my tongs

With a yank you fold the shovel head back alongside the tong's body.  Then you unlock the arms, readying them for use as tongs once more.

> 
[forge]>pound fan on anvil with my forging hammer

Following solid heating in the forge, you carefully fold the glowing red metal back onto itself and hammer it until both ends are fused.  Blow after blow rings sound without any problems at all.
Roundtime: 8 sec.
`

Tested and confirms the adjusting is working flawlessly with the **adjustable_tongs: true** yaml setting in place, then when set to false, the prior behavior swapping to shovel was noted.

I have not tested smelt.lic yet.

Nice job! :)

**Comment by MahtraDR** [comment]
Closes https://github.com/rpherbig/dr-scripts/issues/2161

**Comment by MahtraDR** [comment]
I appreciate the effort at removing the includes, but we have a lot of prefixes missing if we do that. 

image

In red is `DRC`, in orange is `DRCC` in pink is `DRCA` and so on.  I have a VSCode addon I use to highlight like that, if you want to use it.

**Comment by Raykyn55** [comment]
Changed everything but the get_item/stow_item, as it references a method in the script. Should thoose methods be removed, and the get_item be changed to DRCC.get_crafting_item/stow?

**Comment by Raykyn55** [comment]
Ok, did a bunch of work this morning on both forge and smelt, cut some fat, sped up grinding, added your prefixes, split fuel into two methods. I'm posting here, but wondering if i should start a fresh one?  Let me know.

File: forge.lic [hunk 6]
```diff
@@ -2,7 +2,7 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#forge
 =end
 
-custom_require.call(%w[common common-arcana common-crafting common-travel events drinfomon])
+custom_require.call(%w[common common-arcana common-crafting common-money common-travel events drinfomon])
```
Comment (Raykyn55) [inline forge.lic:5]: common-money was added mistakenly, forgot to remove.

**Comment by MahtraDR** [comment]
This depends on https://github.com/rpherbig/dr-scripts/pull/5767

**Comment by asechrest** [comment]
> OK, so I can make this a yaml setting, but didn't for a couple reasons (though absolutely willing to go that route)

> 2. The yaml is already pretty big, and I didn't want to add more niche settings where something like this could suffice (IMO)

Note that I'm working on a substantial set of changes that will help users run help output on any script and see the yaml settings used in the script. You're right that `base.yaml` is becoming unwieldy, so that should help.

**Review by MahtraDR — APPROVED** [review]
This version I am quite happy with. 

It depends on some changes to common-crafting which were merged today. 

We'll merge this in 48ish hours to give those changes a chance to propagate.

**Comment by Dartellum** [comment]
#5065 has this already.

**Review by MahtraDR — APPROVED** [review]
Tested over the last few days. Looks good.

**Pull request merged**